src/main/java/cd/go/contrib/elasticagent/KubernetesInstanceFactory.java

```diff
@@ -227,6 +247,40 @@ private KubernetesInstance createUsingPodYaml(CreateAgentRequest request, Plugin
         return createKubernetesPod(client, elasticAgentPod);
     }
 
+    private KubernetesInstance createUsingRemoteFile(CreateAgentRequest request, PluginSettings settings, KubernetesClient client, PluginRequest pluginRequest) {
+        String fileToDownload = request.properties().get(REMOTE_FILE.getKey());
+        String fileType = request.properties().get(REMOTE_FILE_TYPE.getKey());
+
+        Pod elasticAgentPod = new Pod();
+        ObjectMapper mapper;
+        if ("json".equalsIgnoreCase(fileType)) {
+            mapper = new ObjectMapper(new JsonFactory());
+        }
+        else if ("yaml".equalsIgnoreCase(fileType)) {
+            mapper = new ObjectMapper(new YAMLFactory());
+        }
+        else {
+            throw new IllegalArgumentException("RemoteFileType value should be one of `json` or `yaml`.");
+        }
+
+        File podSpecFile = new File(String.format("pod_spec_%s", UUID.randomUUID().toString()));
+        try {
+            FileUtils.copyURLToFile(new URL(fileToDownload), podSpecFile);
+            LOG.debug(format("Finished downloading %s to %s", fileToDownload, podSpecFile));
+            String spec = FileUtils.readFileToString(podSpecFile, UTF_8);
+            elasticAgentPod = mapper.readValue(spec, Pod.class);
```

Comment:
Wouldn't that depend on the remote spec having POD_POSTFIX and CONTAINER_POSTFIX? If the remote spec doesn't have it, then creating EAs for multiple jobs simultaneously will fail, right? I'm asking so that we can document that the spec users check-in somewhere should conform to some template.

Reply:
That is correct, on the remote spec user has to define a spec with `POD_POSTFIX`. 

> If you have a better idea than this, like maybe replacing the pod name always, please put it up here.

@varshavaradarajan - I thought about it and I am fine with removing `POD_POSTFIX` and make it implicite in code. 

e.g Pod name coming from yaml will became `pod name` + `UUID`.

Reply:
@bdpiparva - for both yaml and remote file spec, I'll check for the presence of POD_POSTFIX and CONTAINER_POSTFIX, if not present, then replace both with name + UUID. Is that alright?

Reply:
@bdpiparva - just did the above for POD_POSTFIX. Because in k8s, pod is the smallest unit to work with. For the container created within the pod, the name doesn't matter.